src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -1755,6 +1755,32 @@ void ScriptMgr::OnPetInitStatsForLevel(Pet* pet)
     FOREACH_SCRIPT(PlayerScript)->OnPetInitStatsForLevel(pet);
 }
 
+bool ScriptMgr::CustomPetAllowedForDeathKnight(Player* player)
+{
+    bool ret = false;
+
+    FOR_SCRIPTS_RET(PlayerScript, itr, end, ret) // return false as default
+        if (itr->second->CustomPetAllowedForDeathKnight(player))
+            ret = true; // change ret value only if at least one script returns true
```

Comment:
as soon as we find one script, we can already `return true;`

I guess there is no need to use `bool ret = false;`

just replace line 1764 with `return true` and line 1766 with `return false`

Reply:
ok, I changed this part